lib/src/rules/prefer_contains.dart

```diff
@@ -0,0 +1,207 @@
+            null &&
+        !type.element.type.isSubtypeOf(typeProvider.stringType)) {
+      return;
+    }
+
+    AstNode search = lengthAccess;
+    while (
+        search != null && search is Expression && search is! BinaryExpression) {
+      search = search.parent;
+    }
+
+    if (search is! BinaryExpression) {
+      return;
+    }
+    BinaryExpression binaryExpression = search;
+
+    Token operator = binaryExpression.operator;
+    DeclaredVariables declaredVariables = context.declaredVariables;
+
+    // Comparing constants with result of indexOf.
+
+    ConstantVisitor visitor = new ConstantVisitor(
+        new ConstantEvaluationEngine(typeProvider, declaredVariables,
+            typeSystem: typeSystem),
+        new ErrorReporter(
+            AnalysisErrorListener.NULL_LISTENER, rule.reporter.source));
+
+    DartObjectImpl rightValue = binaryExpression.rightOperand.accept(visitor);
+    if (rightValue?.type?.name == "int") {
+      // Constant is on right side of comparison operator
+      _checkConstant(binaryExpression, rightValue.toIntValue(), operator.type);
+      return;
+    }
+
+    DartObjectImpl leftValue = binaryExpression.leftOperand.accept(visitor);
+    if (leftValue?.type?.name == "int") {
+      // Constants is on left side of comparison operator
+      _checkConstant(binaryExpression, leftValue.toIntValue(),
+          _invertedTokenType(operator.type));
+    }
+  }
+
+  void _checkConstant(Expression expression, int value, TokenType type) {
+    if (value == -1) {
+      if (type == TokenType.EQ_EQ ||
+          type == TokenType.BANG_EQ ||
+          type == TokenType.LT_EQ ||
+          type == TokenType.GT) {
+        rule.reportLintWithDescription(expression, useContains);
+      } else if (type == TokenType.LT) {
+        // indexOf < -1 is always false
+        rule.reportLintWithDescription(expression, alwaysFalse);
+      } else if (type == TokenType.GT_EQ) {
+        // indexOf >= -1 is always true
+        rule.reportLintWithDescription(expression, alwaysTrue);
+      }
+    } else if (value == 0) {
+      // 'indexOf >= 0' is same as 'contains',
+      // and 'indexOf < 0' is same as '!contains'
+      if (type == TokenType.GT_EQ || type == TokenType.LT) {
+        rule.reportLintWithDescription(expression, useContains);
+      }
+    } else if (value < -1) {
+      // 'indexOf' is always >= -1, so comparing with lesser values makes
+      // no sense.
+      if (type == TokenType.EQ_EQ ||
+          type == TokenType.LT_EQ ||
+          type == TokenType.LT) {
+        rule.reportLintWithDescription(expression, alwaysFalse);
+      } else if (type == TokenType.BANG_EQ ||
+          type == TokenType.GT_EQ ||
+          type == TokenType.GT) {
+        rule.reportLintWithDescription(expression, alwaysTrue);
+      }
+    }
+  }
+
+  TokenType _invertedTokenType(TokenType type) {
+    switch (type) {
+      case TokenType.LT_EQ:
```

Comment:
Is this meant to represent negation? If so 
* `!(<=)` is `>`.
* `!(>=)` is `<`.